### ISSUE TITLE
[mongodb-replicaset] found invalid field args for v1.ExecAction

### ISSUE THREAD

**Issue by scottcrespo** [helm/charts#4706]
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:

**Helm Version**
```
Client: &version.Version{SemVer:"v2.7.2", GitCommit:"8478fb4fc723885b155c924d1c8c410b7a9444e6", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.7.2", GitCommit:"8478fb4fc723885b155c924d1c8c410b7a9444e6", GitTreeState:"clean"}
```

**Kubernetes Version**
```
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.5", GitCommit:"cce11c6a185279d037023e02ac5249e14daa22bf", GitTreeState:"clean", BuildDate:"2017-12-07T18:09:07Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.6", GitCommit:"6260bb08c46c31eea6cb538b34a9ceb3e406689c", GitTreeState:"clean", BuildDate:"2017-12-21T06:23:29Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**:

mongodb-replicaset version 3.0.0

**What happened**:

```
$ helm install stable/mongodb-replicaset
> Error: error validating "": error validating data: found invalid field args for v1.ExecAction
```

**What you expected to happen**:

Chart successfully installs

**How to reproduce it** (as minimally and precisely as possible):

`$ helm install stable/mongodb-replicaset`

**Anything else we need to know**:

Also downloaded the values.yaml file, uncommented resources section, and ran

`$ helm install -f values.yaml stable/mongodb-replicaset`

Still encountered the same problem.

@unguiculus

**Comment by unguiculus** [comment]
The chart installed and tests ran successfully on our CI. I also run it without any problems on my cluster. These clusters have 1.9.x and Helm 2.8.2. Maybe it's related to your Helm or Kubernetes version. I'm kind of clueless from looking at the manifest.

Do you have any idea?

**Comment by rodcloutier** [comment]
I have the same problem when using helm 2.7.2, works fine with 2.8.2

**Comment by zhaoshan86** [comment]
Same problem when using Kubernetes 1.8.2 and Helm 2.7.2

**Comment by JennyLi90** [comment]
I have the same problems uing helm 2.7.0 and Kubernetes 1.9.3. Is there any work around to fix that?

**Comment by unguiculus** [comment]
This might be due to #3331 which was necessary for Kubernetes 1.10 compatibility. 

I guess there are only two options: Use an older version of the chart or upgrade Helm. I would definitely recommend the latter if possible.

cc @mattfarina

**Comment by zhaoshan86** [comment]
Hi @unguiculus ，I found that a property called "args" is used in livenessProbe of mongo container. You can see it in the picture below:

![image](https://user-images.githubusercontent.com/18571265/38969865-dbbeaa50-43c4-11e8-9ca0-564db7e6cdc4.png)

I checked api-reference, "args" is not supported in version 1.8, 1.9 and 1.10. After I removed "args", the chart works fine in kubernetes 1.8 and helm 2.7.2 . 

![image](https://user-images.githubusercontent.com/18571265/38970174-8b6b583a-43c6-11e8-876e-9b6e0c830834.png)

I think maybe higher version helm can ignore properties not supported when generating yaml files. But you still should remove "args" appear in  livenessProbe  and readinessProbe.

**Comment by unguiculus** [comment]
I was wrong in my above comment. Anyways, the issue was fixed in #5008.
